forc/src/ops/forc_build.rs

```diff
@@ -69,18 +69,20 @@ pub fn build(command: BuildCommand) -> Result<pkg::Compiled> {
         } else {
             e
         };
-        info!("  Creating a new `Forc.lock` file. (Cause: {})", cause);
         let plan = pkg::BuildPlan::new(&manifest, SWAY_GIT_TAG, offline)?;
-        let lock = Lock::from_graph(plan.graph());
-        let diff = lock.diff(&old_lock);
-        lock::print_diff(&manifest.project.name, &diff);
-        let string = toml::ser::to_string_pretty(&lock)
-            .map_err(|e| anyhow!("failed to serialize lock file: {}", e))?;
-        fs::write(&lock_path, &string).map_err(|e| anyhow!("failed to write lock file: {}", e))?;
-        info!("   Created new lock file at {}", lock_path.display());
+        info!("  Creating a new `Forc.lock` file. (Cause: {})", cause);
+        create_new_lock(&plan, &old_lock, &manifest, &lock_path)?;
```

Comment:
Rather than creating the lock file here and then potentially a second time after checking the diff, perhaps we can instead keep track of the new lock file cause and create it in one place? E.g. something like this:

```rust
    let mut new_lock_cause = None;
    let mut plan = plan_result.or_else(|e| -> Result<pkgs::BuildPlan> {
        if locked {
            bail!(
                "The lock file {} needs to be updated but --locked was passed to prevent this.",
                lock_path.to_string_lossy()
            );
        }
        new_lock_cause = if e.to_string().contains("No such file or directory") {
            Some(anyhow!("lock file did not exist"))
        } else {
            Some(e)
        };
        let plan = pkg::BuildPlan::new(&manifest, SWAY_GIT_TAG, offline)?;
        Ok(plan, cause)
    });

    let diff = plan.validate(&manifest, SWAY_GIT_TAG)?;
    if !diff.added_is_empty() || !diff.removed.is_empty() {
        plan = plan.apply_new_manifest(diff, SWAY_GIT_TAG, offline)?;
        new_lock_cause = Some(anyhow!("lock file did not match manifest `[dependencies]`"));
    }

    if let Some(cause) = new_lock_cause {
        info!("  Creating a new `Forc.lock` file. (Cause: {})", cause);
        create_new_lock(&plan, &old_lock, &manifest, &lock_path)?;
        info!("   Created new lock file at {}", lock_path.display());
    }
```